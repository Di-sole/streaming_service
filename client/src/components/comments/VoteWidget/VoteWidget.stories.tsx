import { Meta, StoryObj } from '@storybook/react';
import './VoteWidget.module.scss';
import VoteWidget from './VoteWidget';

const meta: Meta<typeof VoteWidget> = {
    title: 'Badges/Vote',
    component: VoteWidget,
};

export default meta;

type Story = StoryObj<typeof VoteWidget>;

export const Vote: Story = {};
