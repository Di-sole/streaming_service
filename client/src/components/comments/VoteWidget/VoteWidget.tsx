import Image from 'next/image';
import TransparentButton from '../../UI/buttons/TransparentButton/TransparentButton';
import like from '../../../../public/icons/like.png';
import dislike from '../../../../public/icons/dislike.png';
import styles from './VoteWidget.module.scss';

const VoteWidget = ({ votes = 10 }) => {
    return (
        <div className={styles.container} data-testid={'voteWidget'}>
            <TransparentButton textColor="bright" className={[styles.button, styles.button_dislike].join(' ')}>
                <Image className={styles.button__image} src={dislike} alt="dislike" />
            </TransparentButton>
            <div className={styles.container__votes}>{votes}</div>
            <TransparentButton textColor="bright" className={[styles.button, styles.button_like].join(' ')}>
                <Image className={styles.button__image} src={like} alt="like" />
            </TransparentButton>
        </div>
    );
};

export default VoteWidget;
