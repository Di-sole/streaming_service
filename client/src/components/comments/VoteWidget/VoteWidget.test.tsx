import { render, screen } from '@testing-library/react';
import VoteWidget from './VoteWidget';

describe('VOTEWIDGET SNAPSHOTS TESTS', () => {
    test('VoteWidget should not be changed', () => {
        render(<VoteWidget />);

        const id = screen.getByTestId('voteWidget');
        expect(id).toMatchSnapshot();
    });
});
