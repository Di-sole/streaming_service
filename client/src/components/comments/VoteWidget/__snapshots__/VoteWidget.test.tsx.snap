// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VOTEWIDGET SNAPSHOTS TESTS VoteWidget should not be changed 1`] = `
<div
  class="container"
  data-testid="voteWidget"
>
  <button
    class="button button_bright button button_dislike"
    type="button"
  >
    <img
      alt="dislike"
      class="button__image"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
  <div
    class="container__votes"
  >
    10
  </div>
  <button
    class="button button_bright button button_like"
    type="button"
  >
    <img
      alt="like"
      class="button__image"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</div>
`;
